### DISCUSSION TITLE
Should meta query hook be added to go with item query hook?

### DISCUSSION THREAD

**Discussion by benosman** [directus/directus#15613] [Feature Requests]
I'm happy to see the `item.query` filter hook added recently, as it is something I have a need for and have had to patch directus up to now.

#10551 

One thing I realised from my own patched implementation, is that if you allow changes to the query object, then there is a risk that the meta.filterCount can mismatch the true item count after the query is modified.

Should there be a corresponding meta.query filter aswell or should the modified query object find its way to the meta service?

On a related note, thie `item.query` hook does not follow the convention of the other item hooks in also allowing `<collection>.item.query` - I can see that being handy if you want to limit to a specific collection.

**Comment by benosman** [comment]
I would be happy to create a pull request for a meta.query filter, but I would like to get an idea if this would be considered.

**Reply by rijkvanzanten** [reply]
The aggregation feature set actually supersedes the meta query flag, so we're actively considering dropping it in the next major release 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
